src/pt/animeq/src/eu/kanade/tachiyomi/animeextension/pt/doramogo/AnimeQ.kt

```diff
@@ -0,0 +1,256 @@
+                    "SD" -> "360p"
+                    "HD" -> "720p"
+                    "FHD" -> "1080p"
+                    else -> format
+                }
+                val iframeSrc = it.selectFirst("iframe")?.tryGetAttr("data-litespeed-src", "src")
+                if (!iframeSrc.isNullOrBlank()) {
+                    return@parallelCatchingFlatMapBlocking getVideosFromURL(iframeSrc, quality)
+                }
+                it.select("script").mapNotNull {
+                    var javascript = it.attr("src")
+                        ?.substringAfter(";base64,")
+                        ?.substringBefore('"')
+                        ?.let { String(Base64.decode(it, Base64.DEFAULT)) }
+
+                    if (javascript.isNullOrBlank()) {
+                        javascript = it.data()
+                    }
+
+                    if (javascript.isNullOrBlank() || "file:" !in javascript) {
+                        return@mapNotNull null
+                    }
+
+                    val videoUrl = javascript.substringAfter("file:\"").substringBefore('"')
+
+                    Video(videoUrl, quality, videoUrl)
+                }
+            }
+    }
+
+    private val bloggerExtractor by lazy { BloggerExtractor(client) }
+    private fun getVideosFromURL(url: String, quality: String?): List<Video> {
+        return when {
+            "blogger.com" in url -> bloggerExtractor.videosFromUrl(url, headers)
+            else -> emptyList()
+        }
+    }
+
+    override fun videoListSelector(): String {
+        throw UnsupportedOperationException()
+    }
+
+    override fun videoFromElement(element: Element): Video {
+        throw UnsupportedOperationException()
+    }
+
+    override fun videoUrlParse(document: Document): String {
+        throw UnsupportedOperationException()
+    }
+
+    // ============================== Settings ==============================
+    override fun setupPreferenceScreen(screen: PreferenceScreen) {
+        ListPreference(screen.context).apply {
+            key = PREF_QUALITY_KEY
+            title = PREF_QUALITY_TITLE
+            entries = PREF_QUALITY_VALUES
+            entryValues = PREF_QUALITY_VALUES
+            setDefaultValue(PREF_QUALITY_DEFAULT)
+            summary = "%s"
+            setOnPreferenceChangeListener { _, newValue ->
+                val selected = newValue as String
+                val index = findIndexOfValue(selected)
+                val entry = entryValues[index] as String
+                preferences.edit().putString(key, entry).commit()
+            }
+        }.also(screen::addPreference)
+    }
+
+    override fun List<Video>.sort(): List<Video> {
+        val quality = preferences.getString(PREF_QUALITY_KEY, PREF_QUALITY_DEFAULT)!!
+        return sortedWith(
+            compareBy(
+                { it.quality.contains(quality) },
+                { Regex("""(\d+)p""").find(it.quality)?.groupValues?.get(1)?.toIntOrNull() ?: 0 },
```

Comment:
Move the `Regex` to a class instance to avoid repeated initialization

Reply:
Done